src/_nebari/stages/infrastructure/template/aws/modules/kubernetes/main.tf

```diff
@@ -30,8 +70,8 @@ resource "aws_eks_node_group" "main" {
   subnet_ids      = var.node_groups[count.index].single_subnet ? [element(var.cluster_subnets, 0)] : var.cluster_subnets
 
   instance_types = [var.node_groups[count.index].instance_type]
-  ami_type       = var.node_groups[count.index].gpu == true ? "AL2_x86_64_GPU" : "AL2_x86_64"
-  disk_size      = 50
+  ami_type       = var.node_groups[count.index].custom_ami != null ? "CUSTOM" : (var.node_groups[count.index].gpu == true ? "AL2_x86_64_GPU" : "AL2_x86_64")
+  disk_size      = var.node_prebootstrap_command == null && var.node_groups[count.index].custom_ami == null ? 50 : null
```

Comment:
There is nothing wrong with the way it is right now, but it would be better if `"AL2_x86_64_GPU"`  and `"AL2_x86_64"` were set as defaults in the python schema itself, so that we could clean this conditionals a bit

Reply:
@viniciusdc 
`ami_type` (i.e. `AL2_x86_64_GPU`/`AL2_x86_64`) is not fetched from nebari-config.yaml or otherwise processed in `__init__.py` in either the current state of Nebari or in the scope of the proposed PR.